src/pynwb/io/base.py

```diff
@@ -59,6 +59,8 @@ def timestamps_carg(self, builder, manager):
         if tstamps_builder is None:
             return None
         if isinstance(tstamps_builder, LinkBuilder):
+            if tstamps_builder.builder.parent is None:
```

Comment:
Why does this conditional get satisfied?

Reply:
I think that this info comes ultimately from `__read_group` and `__read_dataset` on `HDF5IO`.  `__read_group` does not explicitly endow the `LinkBuilder` with a parent and `__read_dataset` does not set one on the builder it returns.

This seems like OK behavior, but I'm not familiar enough with the intended design to tell if it is a bug. It is turning up problems in timestamps custom constructor arg but not elsewhere (though the parent is still None).

Reply:
The only builder that does not have a parent should be a top-level builder i.e. the NWBFile group builder.

Reply:
I suppose it's not totally clear to me why this is true for linked datasets - their parent lives in potentially another file, and might not be itself linked (this seems like a special case of the general expectation that links break the tree structure). What needs to be done with that parent?

I obviously know a lot less about the intended behavior here than you do. Could you explain more about why this behavior seems wrong and what I ought to do in order to fix it?